pandas/tests/frame/test_analytics.py

```diff
@@ -466,6 +466,33 @@ def test_corrwith_mixed_dtypes(self):
         expected = pd.Series(data=corrs, index=['a', 'b'])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.xfail
```

Comment:
Perhaps I'm missing something, but why is this xfailed?

Reply:
I had xfailed this because master was giving an error when columns were duplicated (`ValueError: cannot reindex from a duplicate axis`)

Reply:
Could you rewrite the test to not have duplicate labels?

But I notice now that on master we do allow duplicates in `DataFrame.corrwith`

```python
In [6]: df = pd.DataFrame([[1, 2], [3, 4]], columns=[0, 0], index=[1, 1])

In [7]: df.corrwith(df)
Out[7]:
0    1.0
0    1.0
dtype: float64
```

so we may need to adjust the implementation to not regress on that. It'd be good to add some tests for that if they aren't already present.

Reply:
Hmm, I wonder why that code sample works but this one gives an error:

```
import numpy as np
import pandas as pd

df1 = pd.DataFrame(np.random.random(size=(10, 2)), columns=["a", "b"])
df2 = pd.DataFrame(np.random.random(size=(10, 2)), columns=["a", "a"])

df1.corrwith(df2)
```

Reply:
@TomAugspurger By duplicate labels do you mean the values within the `DataFrame` themselves (i.e., not the indices or column names)?